Key Learning Outcomes
---------------------

After completing this practical the trainee should be able to:

-   Perform a simple NGS data alignment task, with Bowtie2, against one
    interested reference data

-   Interpret and manipulate the mapping output using SAMtools

-   Visualise the alignment via a standard genome browser, e.g. IGV
    browser

Resources Youâ€™ll be Using
-------------------------

### Tools Used

Bowtie2:  
http://bowtie-bio.sourceforge.net/bowtie2/index.shtml

Samtools:  
http://broadinstitute.github.io/picard

BEDTools:  
http://code.google.com/p/bedtools/

UCSC tools:  
http://hgdownload.cse.ucsc.edu/admin/exe/

IGV genome browser:  
http://www.broadinstitute.org/igv/

Useful Links
------------

SAM Specification:  
http://samtools.sourceforge.net/SAM1.pdf

Explain SAM Flags:  
https://broadinstitute.github.io/picard/explain-flags.html

### Sources of Data

http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-11431

## Author Information

*Primary Author(s):*
    Myrto Kostadima kostadim@ebi.ac.uk
   
*Contributor(s):*
    Xi (Sean) Li sean.li@csiro.au

-----------------------

Practice Makes Perfect!
-----------------------

In the chipseq folder you will find the file `gfp.fastq`. Follow the
above described analysis, from the bowtie2 alignment step, for this
dataset as well. You will need these files for the ChIP-Seq module.
